### ISSUE TITLE
Race condition on creating a full chain for a certificate

### ISSUE THREAD

**Issue by aaomidi** [google/certificate-transparency-go#1255]
I've come across a race with how https://github.com/google/certificate-transparency-go/blob/a82fc3ecb947efaa455dd7bc0081f1f1128d1c33/submission/distributor.go#L254-L258 builds the `rootedChain`.

`ctfe.ValidateChain` takes in the _global_ root pool. This pool contains every single root that the application started with. Those get populated here: https://github.com/google/certificate-transparency-go/blob/a82fc3ecb947efaa455dd7bc0081f1f1128d1c33/submission/distributor.go#L163-L170

The race condition is that `PEMCertPool` gets populated in an arbitrary order since the loop is over a map.

`ctfe.ValidateChain` builds various chains, and picks the first valid one: https://github.com/google/certificate-transparency-go/blob/a82fc3ecb947efaa455dd7bc0081f1f1128d1c33/trillian/ctfe/cert_checker.go#L166-L173 

However, not every CT log necessarily has the same trust store, and the RootCompatible method will filter that CT log out.

**Comment by roger2hk** [comment]
@aaomidi Is this issue fully addressed by #1258?

**Comment by aaomidi** [comment]
Not at all. That PR simply disabled this functionality to work around the bug. 

The real solution requires making a trust anchor per log endpoint, or potentially getting rid of this feature entirely.
